### PR TITLE
ci: Fix failing CI due to missing platform interface dependencies

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #835

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

File: packages/flutter/pubspec.yaml [hunk 1]
```diff
@@ -36,3 +36,4 @@ dev_dependencies:
   flutter_test:
     sdk: flutter
   mockito: ^5.2.0
+  path_provider_platform_interface: ^2.0.6
```
Comment (mbfakourii) [inline packages/flutter/pubspec.yaml:39]: Welcome to the Parse Flutter SDK review team ❤️
and i tested the code with this package with the following command
```dart
flutter pub publish --dry-run
```
One of the errors has been solved, but the other error remains, I suggest you add the following package
```
plugin_platform_interface: ^2.1.4
```

**Review by mbfakourii — CHANGES_REQUESTED** [review]

File: packages/flutter/pubspec.yaml (hunk 1 above)
Comment (Nidal-Bakir) [reply packages/flutter/pubspec.yaml:39]: it's working now.

**Review by mbfakourii — APPROVED** [review]

**Comment by mtrezza** [comment]
Is this a bug that affects only our CI or also the published SDK for developers? If the latter, what is the bug that can occur for developers?

**Comment by mbfakourii** [comment]
> Is this a bug that affects only our CI or also the published SDK for developers? If the latter, what is the bug that can occur for developers?

I tested the version of ```parse_server_sdk_flutter: ^3.1.3```, there is no problem on the developer side, probably there is a problem in CI.

**Comment by mtrezza** [comment]
Could it be that issues only occur on specific platforms? Looking at the docs of the [path_provider_platform_interface](https://pub.dev/packages/path_provider_platform_interface):
> This interface allows platform-specific implementations of the path_provider plugin, as well as the plugin itself, to ensure they are supporting the same interface.

**Comment by Nidal-Bakir** [comment]
I think it's normal.
Because the **Parse_SDK** is a **package** and not a **plugin**  if you look at the [pubspec.yaml]( https://github.com/flutter/packages/blob/4aedb859de141673f2bd9985bb46eceac407d9be/packages/path_provider/path_provider/pubspec.yaml#L41) in `path_provider` you will see `plugin_platform_interface: ^2.0.0` under the dev_dependencies and dependencies that's because it needed in tests and in production.

We need these plugins in tests so it's safe to put them under dev_dependencies and the users of the package will not be affected.

Because `pub` will NOT take into account the dev_dependencies from our pubspec.yaml
From the [docs](https://dart.dev/tools/pub/dependencies#dev-dependencies) 
>Dev dependencies differ from regular dependencies in that dev dependencies of packages you depend on are ignored

So do we need new releases? No there is no need for it.

Do we need a better way to test using `path_provider`? Yes
we can do that using DI (Dependency Injection) and not call the `path_provider` functions 
```dart
// core_store_directory_io.dart

class CoreStoreDirectory {
  final PathProvider _pathProvider;

  CoreStoreDirectory(this._pathProvider);

  Future<String> getDatabaseDirectory() async {
    return (await _pathProvider.getLibraryDirectory()).path;
  }

  Future<String> getTempDirectory() async {
    return (await _pathProvider.getTemporaryDirectory()).path;
  }



//--> path_provider.dart

 abstract class PathProvider {
  Future<Directory> getTemporaryDirectory();
  Future<Directory> getLibraryDirectory();
  Future<Directory> getApplicationDocumentsDirectory();
}

class PathProviderImpl extends PathProvider {
  @override
  Future<Directory> getApplicationDocumentsDirectory() {
    return path_provider_package.getApplicationDocumentsDirectory();
  }

  @override
  Future<Directory> getLibraryDirectory() {
    return path_provider_package.getLibraryDirectory();
  }

  @override
  Future<Directory> getTemporaryDirectory() {
    return path_provider_package.getTemporaryDirectory();
  }
}

```

**Comment by mtrezza** [comment]
Ok, so can we merge this with commit type `ci`?

**Review by mtrezza — APPROVED** [review]
Ok, so let's merge this with commit type `ci`, meaning no new release.

**Pull request merged**